lib/fswatcher/fswatcher_unsupported.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// +build solaris,!cgo darwin,!cgo
+
+package fswatcher
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/syncthing/syncthing/lib/config"
+	"github.com/syncthing/syncthing/lib/ignore"
+)
+
+type fsWatcher struct {
+}
+
+type Service interface {
+	Serve()
+	Stop()
+	FsWatchChan() <-chan []string
+	UpdateIgnores(ignores *ignore.Matcher)
+}
+
+func NewFsWatcher(cfg config.FolderConfiguration, ignores *ignore.Matcher) (Service, error) {
+	err := fmt.Errorf("Filesystem notifications are not available on "+
+		"%v-%v due to incompatibility of cgo with cross-compilation.",
+		runtime.GOOS, runtime.GOARCH)
+	l.Warnf(fmt.Sprintln(err))
```

Comment:
I also log an error like this in fswatcher.go and only check for existence of the error higher up and do not log it. My reasoning is, that this should pop up when the debug facility for fswatcher is activated. Is the general philosophy that errors should be logged by the method that called the method that produces an error and not by the method actually producing the error?